onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -39,29 +39,114 @@ Status ReshapeFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level, c
   return Status::OK();
 }
 
+
+/**
+Provide check for Reshape Fusion for DistilBert. The following are subgraphs that
+match the pattern for DistilBert
+
+DistilBert reshape pattern:
+             [Root]
+             /   \ _ _ _ _ _ _ _
+         Shape                  \
+           |                  MatMul(w * w)
+    Gather(indices=0)             |
+            \                     |
+        Unsqueeze               Add(w)
+              \                  /
+             Concat     _ _ _ _ /
+                \      /
+                Reshape
+
+                                  - -> Shape
+                                  |
+Check the subgraph that matches [root] -> MatMul(w * w) -> Add(w) -> Reshape.
+*/
+static bool Match_Linear_Subgraph_1(Graph& graph, const Node& concat, const Node& root, const logging::Logger& logger) {
```

Comment:
reshape_fusion.cc shall not appear in this PR. Please merge master into this branch.